walk-in-interview/src/test/webapp/update-job/script_test.js

```diff
@@ -0,0 +1,563 @@
+        return driver.findElement(By.id('duration'))
+            .getAttribute('value')
+            .then((value) => {
+              assert.equal(value, 'ONE_WEEK');
+            });
+      });
+    });
+
+    describe('Job Expiry', () => {
+      it('checks the title text', () => {
+        return driver.findElement(By.id('expiry-title'))
+            .getText()
+            .then((title) => {
+              assert.equal(title, 'Job Expiry');
+            });
+      });
+
+      it('checks the type attribute', () => {
+        return driver.findElement(By.id('expiry'))
+            .getAttribute('type')
+            .then((value) => {
+              assert.equal(value, 'date');
+            });
+      });
+
+      it('checks the min attribute', () => {
+        return driver.findElement(By.id('expiry'))
+            .getAttribute('min')
+            .then((value) => {
+              const expected = new Date().toISOString().substr(0, 10);
+
+              assert.equal(value, expected);
+            });
+      });
+
+      it('checks the max attribute', () => {
+        return driver.findElement(By.id('expiry'))
+            .getAttribute('max')
+            .then((value) => {
+              const date = new Date();
+              date.setFullYear(date.getFullYear() + 1);
+              const expected = date.toISOString().substr(0, 10);
+
+              assert.equal(value, expected);
+            });
+      });
+
+      it('checks if default option properly selected', () => {
+        return driver.findElement(By.id('expiry'))
+            .getAttribute('value')
+            .then((value) => {
+              assert.equal(value, '2020-10-05');
+            });
+      });
+    });
+  });
+
+  describe('Page Functionality Tests', () => {
+    beforeEach(() => {
+      /**
+       * For the functionality tests, since we are testing the
+       * cancel/submit buttons, the page url may have changed
+       * in the test.
+       * Always resets the current page to the job page 
+       * for the rest of the tests.
+       */
+        return driver.get(JOBPAGE_URL);
+    });
+
+    describe('Cancel Button', () => {
+      /**
+       * Clicking the cancel button should return the user to the homepage
+       * and not make any POST request.
+       */
+      it('should return to homepage', () => {
+        return clickCancel(driver)
+            .then(() => driver.wait(until.urlIs(HOMEPAGE_URL)))
+            .then(() => driver.getCurrentUrl())
+            .then((currUrl) => assert.equal(currUrl, HOMEPAGE_URL));
+      });
+    });
+
+    /**
+     * If a field is not valid, then clicking submit will display an error
+     * message with the invalid field, and no POST request will be made.
+     */
+    describe('Submit Button', () => {
+      const date = new Date();
+      const today = `${(date.getMonth() + 1)}-${date.getDate()}` +
+        `-${date.getFullYear()}`;
+
+      beforeEach('add all valid inputs', () => {
```

Comment:
Do you need this? I thought all the fields were already filled in for the update job page.

Reply:
I think it would be better if you tested that the correct error message appeared if the user empties or fills another invalid value into a certain field

Reply:
```Do you need this?``` this is to test the invalid input, not to test if the fields are filled by default value; so i just fill them all again to make sure

i will add a TODO for more invalid value

Reply:
hmm I'm still not too sure about this. I think it would make more sense for you to use the pre-filled in data and then clear or change those fields.